File: problem/Can_Place_Flowers/canPlaceFlowers.c
bool canPlaceFlowers(int *flowerbed, int flowerbedSize, int n)
{
    for (int i = 0; i < flowerbedSize && n > 0; i++)
    {
        bool prev = i == 0 || !flowerbed[i - 1];
        bool fut = i == flowerbedSize - 1 || !flowerbed[i + 1];
        if (prev && fut && !flowerbed[i])
        {
            flowerbed[i] = 1;
            n--;
        }
    }
    return n <= 0;
}

File: problem/Can_Place_Flowers/canPlaceFlowers.py
class Solution(object):
    def canPlaceFlowers(self, flowerbed, n):
        flowerbed_size = len(flowerbed)
        for i in range(flowerbed_size):
            if n <= 0:
                break
            prev = i == 0 or flowerbed[i - 1] == 0
            fut = i == flowerbed_size - 1 or flowerbed[i + 1] == 0
            if prev and fut and flowerbed[i] == 0:
                flowerbed[i] = 1
                n -= 1
        return n <= 0